pilot/pkg/model/sidecar.go

```diff
@@ -231,13 +247,59 @@ func convertIstioListenerToWrapper(ps *PushContext, configNamespace string,
 		ConfigNamespace: configNamespace,
 	}
 
-	out.services = out.selectServices(ps.Services(&dummyNode))
+	out.services = out.selectServices(ps.Services(&dummyNode), configNamespace)
 	meshGateway := map[string]bool{IstioMeshGateway: true}
 	out.virtualServices = out.selectVirtualServices(ps.VirtualServices(&dummyNode, meshGateway))
 
 	return out
 }
 
+// ServiceForHostname returns the service associated with a given hostname following SidecarScope
+func (sc *SidecarScope) ServiceForHostname(hostname Hostname, serviceByHostname map[Hostname]map[string]*Service) *Service {
+	if sc == nil {
```

Comment:
I didn't know you can call a function on a nil object.. til

Reply:
Me too. Probably was a bad idea though, will be cleaned up after my other PR lands

Reply:
lol, actually, golang allows this ops.

Reply:
Under what condition is this nil? Haven’t we merged the pr that establishes a sidecar for everything in the system?

Reply:
When is this nil? Isn’t theee a default sidecar for everything now ?